src/main/java/org/mandas/docker/client/CompressedDirectory.java

```diff
@@ -137,14 +136,14 @@ public void close() throws IOException {
     Files.delete(file);
   }
 
-  static ImmutableList<DockerIgnorePathMatcher> parseDockerIgnore(Path dockerIgnorePath)
+  static List<DockerIgnorePathMatcher> parseDockerIgnore(Path dockerIgnorePath)
       throws IOException {
-    final ImmutableList.Builder<DockerIgnorePathMatcher> matchersBuilder = ImmutableList.builder();
+    final List<DockerIgnorePathMatcher> matchersBuilder = new ArrayList<>();
```

Comment:
Should `new ArrayList<>` be typed as `DockerIgnorePathMatcher` in order to avoid a warning?